src/index.html

```diff
@@ -14,6 +14,87 @@
   </head>
   <body>
     <h1 class="logo">MA</h1>
-    <script src="scripts/main.js"></script>
+    <div id="notifications"></div>
+    <script>
+      function createNotification(message, type) {
+        const notification = document.createElement('div');
+        notification.className = type;
+        notification.dataset.qa = 'notification';
+        notification.textContent = message;
+        document.getElementById('notifications').appendChild(notification);
+      }
+
+      let firstPromiseResolved = false;
+      let secondPromiseResolved = false;
+      let thirdPromiseResolved = false;
+      let leftClicked = false;
+      let rightClicked = false;
+
+      const firstPromise = new Promise((resolve, reject) => {
+        setTimeout(() => {
+          if (!firstPromiseResolved) {
+            reject('First promise was rejected');
+          }
+        }, 3000);
+      });
+
+      const secondPromise = new Promise((resolve) => {});
+
+      const thirdPromise = new Promise((resolve) => {});
+
+      function handleClick(event) {
+        if (event.button === 0) {
+          if (!firstPromiseResolved) {
+            firstPromiseResolved = true;
+            firstPromise
+              .then((message) => createNotification(message, 'success'))
+              .catch((message) => createNotification(message, 'error'));
+            firstPromiseResolved = true;
```

Comment:
The `firstPromiseResolved` flag is set to true twice, which is redundant. You should remove this duplicate assignment.